### PR TITLE
topk: use multicore pad when appropriate

### PR DESCRIPTION
**Pull request by handrewsTT**
### Problem description
When running GPT-OSS-20B on batch=1 we were seeing the console flooded with the warning:
```
warning  |              Op | Only bfloat16 and non-L1 tiled tensors are currently supported for multicore tiled pad. Falling back to 1 core. #29295 (pad_device_operation.cpp:81)
```

When the router logits are in DRAM and bfloat16 it should be safe to use multicore pad according to the warnings.

### What's changed
Use multicore pad when logits are bfloat16 DRAM interleaved.

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/runs/20372806322) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/runs/20372812782) CI with demo tests passes (if applicable)

### CODE REVIEW

**Comment by edwinleeTT** [comment]
It looks like the issue in the warning https://github.com/tenstorrent/tt-metal/issues/29295 is already closed. @ntarafdar should either that warning or that ticket be updated?

It also looks like pad already has a similar check inside it:
https://github.com/tenstorrent/tt-metal/blob/48f52ba82f0259c6c8c63deb72d8998abadf2ee2/ttnn/cpp/ttnn/operations/data_movement/pad/device/pad_device_operation.cpp#L74

Do we need to do this check twice, or is it safe to just set it to true and let pad's checks handle it?
And just to confirm, this causes no issues to the model and topk tests?

**Review by aczajkowskiTT — APPROVED** [review]

**Review by bbradelTT — APPROVED** [review]
